dowhy/causal_refuters/non_parametric_sensitivity_analyzer.py

```diff
@@ -0,0 +1,270 @@
+                self.r2yu_tw = 1
+                self.logger.warning(
+                    "Warning: r2yu_tw can not be > 1. Try a lower effect_fraction_on_outcome. Setting r2yu_tw to 1"
+                )
+        else:
+            self.r2yu_tw = self.effect_strength_outcome
+            self.r2tu_w = self.effect_strength_treatment
+
+        benchmarking_results = self.perform_benchmarking(
+            r2yu_tw=self.r2yu_tw,
+            r2tu_w=self.r2tu_w,
+            significance_level=self.significance_level,
+            is_partial_linear=self.is_partial_linear,
+        )
+        self.results = pd.DataFrame(benchmarking_results, index=[0])
+        self.RV = self.calculate_robustness_value(alpha=None, is_partial_linear=self.is_partial_linear)
+        self.RV_alpha = self.calculate_robustness_value(
+            alpha=self.significance_level, is_partial_linear=self.is_partial_linear
+        )
+        if plot == True:
+            self.plot()
+
+        return self
+
+    def get_phi_lower_upper(self, Cg, Calpha):
+        """
+        Calculate lower and upper influence function (phi)
+
+        :param Cg: measure of strength of confounding that omitted variables generate in outcome regression
+        :param Calpha: measure of strength of confounding that omitted variables generate in treatment regression
+
+        :returns : lower bound of phi, upper bound of phi
+        """
+
+        bounds_estimator_upper = self.neyman_orthogonal_score_theta + ((Cg * Calpha) / (2 * self.S)) * (
+            self.sigma_2 * self.neyman_orthogonal_score_treatment + self.nu_2 * self.neyman_orthogonal_score_treatment
+        )
+        bounds_estimator_lower = self.neyman_orthogonal_score_theta - ((Cg * Calpha) / (2 * self.S)) * (
+            self.sigma_2 * self.neyman_orthogonal_score_treatment + self.nu_2 * self.neyman_orthogonal_score_treatment
+        )
+
+        return bounds_estimator_lower, bounds_estimator_upper
+
+    def get_alpharegression_var(self, X, numeric_features, split_indices, reisz_model=None):
+        """
+        Calculates the variance of reisz function
+
+        :param X: numpy array containing set of regressors
+        :param split_indices: training and testing data indices obtained after cross folding
+
+        :returns: variance of reisz function
+        """
+        if reisz_model is None:
+            reisz_model = get_alpha_estimator(
+                cv=split_indices,
+                X=X,
+                max_degree=self.reisz_polynomial_max_degree,
+                estimator_list=self.alpha_s_estimator_list,
+                estimator_param_list=self.alpha_s_estimator_param_list,
+                numeric_features=numeric_features,
+                plugin_reisz=self.plugin_reisz,
+            )
+
+        num_samples = X.shape[0]
+        pred = np.zeros(num_samples)
+        if self.plugin_reisz:
+            for train, test in split_indices:
+                reisz_fn_fit = reisz_model.fit(X[train])

```

Comment:
When we call a fit() function and fit a model over data, should we be checking the quality of the fit against the target and warning if it has a high error?  E.g., "model didn't converge.  the sensitivity analysis might not be reliable"?  Or is that error rate actually incorporated as part of the sensitivity analysis' logic?

Reply:
the confidence interval for the bound does depend on the error in the alpha/reisz model. That said, there could be other issues with model training (e.g., overfitting). Overall, it does make sense to have basic predictive checks and provide warnings. That can be an enhancement to consider.

Reply:
Let's open an issue or leave a comment in the code to that effect.

Reply:
Added Issue #639